src/libmongoc/tests/test-mongoc-client-side-encryption.c

```diff
@@ -5679,6 +5679,140 @@ test_create_encrypted_collection_bad_keyId (void *unused)
    mongoc_client_destroy (client);
 }
 
+typedef struct listen_socket {
+   mongoc_socket_t *socket;
+   mongoc_cond_t cond;
+   bson_mutex_t mutex;
+   bool failed;
+   char ip[16];
+   unsigned short port;
+} listen_socket_args_t;
+
+static BSON_THREAD_FUN (listen_socket, arg)
+{
+   listen_socket_args_t *args = arg;
+   mongoc_socket_t *socket;
+   struct sockaddr_in server_addr = {0};
+   args->failed = false;
+
+   // create a TcpListener on 127.0.0.1 endpoint
+   socket = mongoc_socket_new (AF_INET, SOCK_STREAM, 0);
+   BSON_ASSERT (socket);
+
+   server_addr.sin_family = AF_INET;
+   server_addr.sin_addr.s_addr = htonl (INADDR_LOOPBACK);
+   server_addr.sin_port = htons (0);
+   mongoc_socklen_t addr_len = (mongoc_socklen_t) sizeof server_addr;
+
+   int r =
+      mongoc_socket_bind (socket, (struct sockaddr *) &server_addr, addr_len);
+   BSON_ASSERT (r == 0);
+
+   // forward the port and ip for mongocryptdURI
+   r = mongoc_socket_getsockname (
+      socket, (struct sockaddr *) &server_addr, &addr_len);
+   BSON_ASSERT (r == 0);
+
+   bson_mutex_lock (&args->mutex);
+   args->port = ntohs (server_addr.sin_port);
+   inet_ntop (AF_INET, &server_addr.sin_addr, args->ip, sizeof (args->ip));
+   mongoc_cond_signal (&args->cond);
+   bson_mutex_unlock (&args->mutex);
+
+   // listen on socket
+   r = mongoc_socket_listen (socket, 100);
+   BSON_ASSERT (r == 0);
+   _mongoc_usleep (1000); // wait to see if received connection
+   mongoc_socket_t *ret = mongoc_socket_accept (socket, 100);
+   if (ret) {
+      // not null received a connection and test should fail
+      bson_mutex_lock (&args->mutex);
+      args->failed = true;
+      mongoc_cond_signal (&args->cond);
+      bson_mutex_unlock (&args->mutex);
+   }
```

Comment:
```suggestion
   if (ret) {
      // not null received a connection and test should fail
      args->failed = true;
   }
   // signal that test is complete.
   bson_mutex_lock (&args->mutex);
   mongoc_cond_signal (&args->cond);
   bson_mutex_unlock (&args->mutex);
```

Signal as soon as the listener thread stops listening. This results in the test speeding up.